Add HLT filter to select online pf taus based on their impact parameter

sarafiorendi opened the pull request:
#### PR description:

A new HLTFilter is added in order to be able to apply cuts on the PF tau impact parameters values as calculated by the PFTauTransverseImpactParameter class .
The purpose is to use this filter in an HLT path for Run3 targeting displaced taus.

sarafiorendi committed:
add filter to cut on tau dxy

apply code format

accept suggestion

Co-authored-by: Marino Missiroli <m.missiroli@cern.ch>

merge .h into .cc file, apply suggestions from review

gather includes

move to RecoTauTag/HLTProducers

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27707

- This PR adds an extra 16KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27707/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27707/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+  desc.add<double>("MaxDxy",  999999.0);
+  desc.add<int>("MinN", 1);
+  desc.add<int>("TriggerType", 84);
+  descriptions.add(defaultModuleLabel<HLTTauDxyFilter>(), desc);
```
```suggestion
  descriptions.addWithDefaultLabel(desc);
```

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+                   << "\tmin no. passing taus: " << m_MinTaus << "\tTriggerType: " << m_TriggerType << std::endl;
+}
+
+HLTTauDxyFilter::~HLTTauDxyFilter() = default;
+
```
```suggestion
```

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+
+  edm::LogInfo("") << " (HLTTauDxyFilter) trigger cuts: " << std::endl
+                   << "\tmin/max dxy value: [" << m_MinDxy << ".." << m_MaxDxy << "]" << std::endl
+                   << "\tmin no. passing taus: " << m_MinTaus << "\tTriggerType: " << m_TriggerType << std::endl;
```
```suggestion
                   << "\tmin no. passing taus: " << m_MinTaus << "\tTriggerType: " << m_TriggerType;
```
I might be misremembering, but I dont think endl has any effect at the end of `LogInfo` (it makes a new line anyway).

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+#include <string>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/Common/interface/Handle.h"
```
```suggestion
```

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+
+
+//
+// constructors and destructor
+//
```
```suggestion
```

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+}
+
+
+// ------------ method called to produce the data  ------------
```
```suggestion
```

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.h:
```diff
@@ -0,0 +1,47 @@
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+
+namespace edm {
+  class ConfigurationDescriptions;
+}
+
+//
+// class declaration
+//
+
```
```suggestion
```

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,81 @@
+
+HLTTauDxyFilter::HLTTauDxyFilter(const edm::ParameterSet& config)
+    : HLTFilter(config),
+      m_Taus(config.getParameter<edm::InputTag>("Taus")),
+      m_TausIP(config.getParameter<edm::InputTag>("TausIP")),
+      m_MinDxy(config.getParameter<double>("MinDxy")),
+      m_MaxDxy(config.getParameter<double>("MaxDxy")),
+      m_MinTaus(config.getParameter<int>("MinN")),
+      m_TriggerType(config.getParameter<int>("TriggerType")) {
+  m_TausToken = consumes<std::vector<reco::PFTau>>(m_Taus), 
+  m_TausIPToken = consumes<edm::AssociationVector<reco::PFTauRefProd,std::vector<reco::PFTauTransverseImpactParameterRef>>>(m_TausIP),
```
```suggestion
      m_TausInputTag(config.getParameter<edm::InputTag>("Taus")),
      m_TausToken(consumes(m_TausInputTag)), 
      m_TausIPToken(consumes(config.getParameter<edm::InputTag>("TausIP"))),
      m_MinDxy(config.getParameter<double>("MinDxy")),
      m_MaxDxy(config.getParameter<double>("MaxDxy")),
      m_MinTaus(config.getParameter<int>("MinN")),
      m_TriggerType(config.getParameter<int>("TriggerType")) {
```

missirol reviewed (commented):
Here's the first round of comments: mostly small technical changes, plus some questions.

"Code-checks" failed, and this needs to be fixed. Please remember to run `scram b code-checks ; scram b code-format`, and commit the suggested changes, before pushing updates to the branch.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27709

- This PR adds an extra 16KB to repository

cmsbuild commented:
A new Pull Request was created by  @sarafiorendi  for master.

It involves the following packages:

- HLTrigger/btau (**hlt**)


@cmsbuild, @missirol, @Martin-Grunewald can you please review it and eventually sign? Thanks.
@AlexDeMoor, @emilbols, @JyothsnaKomaragiri, @silviodonato, @Martin-Grunewald, @missirol, @andrzejnovak, @demuller this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27711

ERROR: Build errors found during clang-tidy run.
```
HLTrigger/btau/plugins/HLTTauDxyFilter.cc:37:18: error: definition of implicitly declared destructor [clang-diagnostic-error]
HLTTauDxyFilter::~HLTTauDxyFilter() = default;
                 ^
Suppressed 1456 warnings (1452 in non-user code, 3 NOLINT, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

sarafiorendi commented:
Hi @missirol,
thanks for the review. I applied almost all of your suggestions, but I still need to answer about if a check for a valid TIP reference is needed.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27712

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - HLTrigger/btau/plugins/HLTTauDxyFilter.h:
    - Added: c82e93921b0a63dda6868315f99ba001f57f7351
    - Modified: d4bae1f0fc008bcb2bd5a86a3ab28f6e31450f5b, 434f3fe3cf41f9921c387eed0e4447779d75bb7b
    - Deleted: 329e5d73c6daa94601ac917ed454757e528a96a9

cmsbuild commented:
Pull request #36666 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,97 @@
+  const int m_TriggerType;
+};
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
```
```suggestion
```
Please move these header files upstairs with the other ones. I'm not sure `RefToBase.h` is needed, I would try to remove it.

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,97 @@
+  return accept;
+}
+
+#include "FWCore/Framework/interface/MakerMacros.h"
```
```suggestion
```
This one can also go upstairs.

missirol commented on HLTrigger/btau/plugins/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,97 @@
+ *
+ */
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+
+#include <vector>
+#include <string>
```
```suggestion
#include "FWCore/Framework/interface/Frameworkfwd.h"
#include "FWCore/Framework/interface/MakerMacros.h"
#include "FWCore/MessageLogger/interface/MessageLogger.h"
#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
#include "FWCore/Utilities/interface/InputTag.h"
#include "FWCore/Utilities/interface/EDGetToken.h"
#include "DataFormats/TauReco/interface/PFTau.h"
#include "DataFormats/TauReco/interface/PFTauFwd.h"
#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
#include "HLTrigger/HLTcore/interface/HLTFilter.h"

#include <vector>
```

missirol reviewed (commented):
Thanks @sarafiorendi , I will wait for you to check that.

One more suggestion below to adjust the `#include` statements.

sarafiorendi commented:
Hi Marino,
checks for a valid reference are performed by the PFTauTransverseImpactParameters class here [1] and [2] so I think a further check in this new filter is not needed.

[1]
https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/plugins/PFTauTransverseImpactParameters.cc#L84
[2]
https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/plugins/PFTauTransverseImpactParameters.cc#L135

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27717

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - HLTrigger/btau/plugins/HLTTauDxyFilter.h:
    - Added: c82e93921b0a63dda6868315f99ba001f57f7351
    - Modified: d4bae1f0fc008bcb2bd5a86a3ab28f6e31450f5b, 434f3fe3cf41f9921c387eed0e4447779d75bb7b
    - Deleted: 329e5d73c6daa94601ac917ed454757e528a96a9

cmsbuild commented:
Pull request #36666 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented:
> Hi Marino, checks for a valid reference are performed by the PFTauTransverseImpactParameters class here [[1]](https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/plugins/PFTauTransverseImpactParameters.cc#L84) and [[2]](https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/plugins/PFTauTransverseImpactParameters.cc#L135) so I think a further check in this new filter is not needed.

@sarafiorendi, thanks for the info. I see your point, but I think there is still a potential issue if there is a (python) misconfiguration such that the two collections of Taus differ, one being the collection used as "key" for the `AssociationVector` in `h_TausTIP` (i.e. the `reco::PFTauRefProd`), and the other being the one in `h_Taus`.

I think the call in [`h_TausTIP->value` calls a vector element by index](https://github.com/cms-sw/cmssw/blob/0c56c1e31f0325a2ffef3fead90538e232b25a6d/DataFormats/Common/interface/AssociationVector.h#L102) (`tauref.key()`), and that can lead to a seg-fault if the two collections in question have different sizes. This type of misconfiguration might be unlikely, but I think it's better to guard against it.

Would it be okay to require inside the filter that the two collections (the one in `h_Taus` and the one referenced by the `AssociationVector`) have to be the same?

If yes, I can go ahead and ask fwk experts what's the best way to check/enforce that.

sarafiorendi commented:
Hi Marino, yes I agree it would be better to require the two collections to be the same. If you can ask the experts, I will then implement their suggestion. thanks a lot!

missirol commented:
Hi @makortel, could you please advise on a simple fwk question?

In this PR, we have (inside a `::filter` method) a `edm::Handle<PFTauCollection>` (named `m_Taus`) and a `edm::Handle<edm::AssociationVector<edm::RefProd<PFTauCollection>, [..] >>` (named `m_TausTIP`), and would like to check if `m_Taus` is referencing the same collection as the key of the `AssociationVector`.

Would `m_Taus.id() == m_TausIP->keyProduct().id()` be the correct way to do that? Is there a better way?

sarafiorendi commented:
hi again, 
I've been asked by the Tau POG to move this filter into the RecoTauTag/HLTProducers folder, should I just issue a "git mv"?

perrotta commented:
> hi again, I've been asked by the Tau POG to move this filter into the RecoTauTag/HLTProducers folder, should I just issue a "git mv"?

Ciao Sara.
`git mv` and squash all commits, so that no track remains of the code in the previous folder
(or alternatively just close this PR and make a new one with the code in the correct location, but as such we'll loose from the new PR all comments already posted here)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27764

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #36666 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented:
@sarafiorendi 

> I've been asked by the Tau POG to move this filter into the RecoTauTag/HLTProducers folder

Mh.. then, we need to go one step backwards: right now, the whole class is in one file under `src/*.cc`, which doesn't work (it worked in `btau/plugins/` because that folder has its own `BuildFile.xml`). Since `RecoTauTag/HLTProducers/plugins` does not exist unfortunately, I guess the solution is to split again in two files (`interface/*h` and `src/*cc`), and add the `DEFINE_FWK_MODULE` call in `RecoTauTag/HLTProducers/src/SealModule.cc`. For example, you can follow what is done for the plugin `HLTPFTauPairLeadTrackDzMatchFilter`.

[*] One can argue we should add `RecoTauTag/HLTProducers/plugins`, but I would postpone that to a separate technical PR by HLT. If others have preferences about this (doing it, or not), feel free to let me know.

makortel commented:
> right now, the whole class is in one file under `src/*.cc`, which doesn't work

Why would it not work?

makortel commented:
> Would `m_Taus.id() == m_TausTIP->keyProduct().id()` be the correct way to do that? Is there a better way?

That should be a correct way. The `operator[](KeyRef const&)` does exactly that check
https://github.com/cms-sw/cmssw/blob/77e4d1f7d15dc81d5fb47a182f1d564f02413c73/DataFormats/Common/interface/AssociationVector.h#L164-L170

missirol commented:
> > right now, the whole class is in one file under `src/*.cc`, which doesn't work
> 
> Why would it not work?

I stand corrected, it works (I just didn't remember seeing this before; in retrospect, i think i understand why it should compile without problems). Thanks for both suggestions, Matti.

@sarafiorendi you can ignore my last comment, sorry for the confusion.

missirol commented on RecoTauTag/HLTProducers/src/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,94 @@
+  if (saveTags())
+    filterproduct.addCollectionTag(m_TausInputTag);
+
+  auto const h_TausTIP = event.getHandle(m_TausTIPToken);
```
```suggestion
  auto const& tausTIP = event.get(m_TausTIPToken);
```

missirol commented on RecoTauTag/HLTProducers/src/HLTTauDxyFilter.cc:
```diff
@@ -0,0 +1,94 @@
+  }
+
+  // filter decision
+  bool accept = (nTau >= m_MinTaus);
+  edm::LogInfo("") << " trigger accept ? = " << accept << " nTau = " << nTau << std::endl;
```
```suggestion
  bool const accept = (nTau >= m_MinTaus);
  edm::LogInfo("") << " trigger accept ? = " << accept << " nTau = " << nTau;
```

missirol reviewed (commented):
This should be the last set of comments.

@sarafiorendi , since the PR tests won't exercise this plugin (other than compiling it), please check that its final version works as intended.

sarafiorendi committed:
apply latest suggestions,move to cut string obj sel

more generic name after change to cut string

apply code format

implement latest suggestions

rename pftau, add const

sarafiorendi commented:
hi again,
I applied all your suggestions, plus I changed it to use a cut string, so that other selections (e.g. on the IP significance) could be applied if needed. For this reason I also changed the name to a more generic one.

cmsbuild commented:
Pull request #36666 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,94 @@
+/** \class HLTTauIPFilter
```
```suggestion
/** \class HLTPFTauTIPFilter
```
To follow the names of the DataFormats more closely, I would suggest to use `HLTPFTauTIPFilter`.

sarafiorendi replied:
actually I chose IP instead of TIP because despite the "PFTauTransverseImpactParameter" name of that class, you can also retrieve the 3D impact parameter [1] and cut on it. This is why I did not the "T" (transverse) in the filter name. 
I don't have strong opinions on this filter name though.

[1] https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/plugins/PFTauTransverseImpactParameters.cc#L144

missirol replied:
That's okay with me (too bad the name of the DataFormat is a bit misleading).

missirol replied:
Should have been clearer: no problem with "IP", but I would still prefer "PFTau".
Could we please use `HLTPFTauIPFilter` since the filter deals with "PFTau"s? (atm, we have other filters/producers in the menu named `HLTPFTau*`, but I couldn't find any called `HLTTau*`)

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,94 @@
+/** \class HLTTauIPFilter
+ *
+ *  This class is an HLTFilter implementing
+ *  a minimum/maximum dxy cut on the tau object
```
I guess this comment now needs to be updated.

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,94 @@
+bool HLTTauIPFilter::hltFilter(edm::Event& event,
+                               const edm::EventSetup& setup,
+                               trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  auto const m_Taus = event.getHandle(m_TausToken);
+  if (saveTags())
+    filterproduct.addCollectionTag(m_TausInputTag);
+
+  auto const& m_TausTIP = event.get(m_TausTIPToken);
```
```suggestion
  auto const h_Taus = event.getHandle(m_TausToken);
  if (saveTags())
    filterproduct.addCollectionTag(m_TausInputTag);

  auto const& tausTIP = event.get(m_TausTIPToken);
```
In this plugin, `m_*` names are used for class members, so shouldn't be used for 'local' objects too (`h_` can help identifying that obj as a Handle).

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,94 @@
+  auto const& m_TausTIP = event.get(m_TausTIPToken);
+
+  int nTau = 0;
+  for (reco::PFTauCollection::size_type iPFTau = 0; iPFTau < m_Taus->size(); iPFTau++) {
+    reco::PFTauRef tauref(m_Taus, iPFTau);
+
+    if ((*m_cut)(*m_TausTIP[tauref])) {
```
```suggestion
  for (reco::PFTauCollection::size_type iPFTau = 0; iPFTau < h_Taus->size(); iPFTau++) {
    reco::PFTauRef tauref(h_Taus, iPFTau);

    if (m_tauTIPSelector(*tausTIP[tauref])) {
```

missirol reviewed (commented):
Comments on the latest changes.

sarafiorendi commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,94 @@
+      m_TausTIPToken;
+
+  const int m_MinTaus;  // min. number of taus required to pass the cuts
+  std::unique_ptr<StringCutObjectSelector<reco::PFTauTransverseImpactParameter>> m_cut;
+  std::string m_cutString;
```
no clear advantage unless I'm missing something. I'll implement your suggestion

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27786

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - RecoTauTag/HLTProducers/src/HLTTauDxyFilter.cc:
    - Added: 91cd41855952750c97a8fc993ec00e75cac7fb36
    - Modified: bc932f5dcec7fb4c8e9c1f547a66c5da430e507d
    - Deleted: 642ee8e6ba3c52832894c86236c34c807406b5c3

cmsbuild commented:
Pull request #36666 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,94 @@
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include <vector>
```
From previous round of comments.

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,92 @@
+      m_TausTIPToken;
+
+  const int m_MinTaus;  // min. number of taus required to pass the cuts
+  std::string m_tauTIPSelectorString;
+  StringCutObjectSelector<reco::PFTauTransverseImpactParameter> m_tauTIPSelector;
```
```suggestion
  const std::string m_tauTIPSelectorString;
  const StringCutObjectSelector<reco::PFTauTransverseImpactParameter> m_tauTIPSelector;
```
From previous round of comments.

missirol commented on RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc:
```diff
@@ -0,0 +1,92 @@
+  for (reco::PFTauCollection::size_type iPFTau = 0; iPFTau < h_Taus->size(); iPFTau++) {
+    reco::PFTauRef tauref(h_Taus, iPFTau);
+
+    if ((m_tauTIPSelector)(*TausTIP[tauref])) {
```
```suggestion
    if (m_tauTIPSelector(*TausTIP[tauref])) {
```
Parentheses not needed.

missirol reviewed (commented):
Last set of comments. Some were missed in the previous round, and are repeated below.

Optional: consider squashing again into 1 commit before the final push.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36666/27788

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - RecoTauTag/HLTProducers/src/HLTTauDxyFilter.cc:
    - Added: 91cd41855952750c97a8fc993ec00e75cac7fb36
    - Deleted: fd3fedb582e24959d72897d396403a3d4372bf90

cmsbuild commented:
Pull request #36666 was updated. @cmsbuild, @missirol, @Martin-Grunewald can you please check and sign again.

missirol commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab50aa/21724/summary.html
**COMMIT**: fd3fedb582e24959d72897d396403a3d4372bf90
**CMSSW**: CMSSW_12_3_X_2022-01-14-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36666/21724/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 43
 - DQMHistoTests: Total histograms compared: 3461659
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3461637
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 42 files compared)
 - Checked 181 log files, 42 edm output root files, 43 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-01-14-1100+ab50aa/47856/triggerResults): no differences found

missirol commented:
+hlt

 - new `HLTFilter` to select `PFTau`s based on their IP properties
 - presently not used in any validation wf (nor any HLT menu)

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

missirol commented:
@sarafiorendi , it would be great if you could update the PR title and description to reflect the latest changes to the plugin (dxy -> IP properties).

qliphy commented:
> @sarafiorendi , it would be great if you could update the PR title and description to reflect the latest changes to the plugin (dxy -> IP properties).

ping @sarafiorendi

qliphy commented:
+1
